### ISSUE TITLE
Unused var error missing when using es6 modules

### ISSUE THREAD

**Issue by dominicbarnes** [eslint/eslint#2678]
It appears that when you use ES6 modules to `export` a function, the `no-unused-var` rule does not run properly on those function arguments. (vars declared in the function body still lint correctly)

```js
function fn1({ a, b }) {
  console.log(a);
}

fn1();

export function fn2({ x, y }) {
  console.log(x);
}

fn2();
```

Given the following config:

```js
{
  "env": {
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "rules": {
    "no-unused-vars": 2
  }
}
```

When I run this in the CLI:

```sh
$ eslint --reset test.js

test.js
  2:18  error  b is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

Notice that `fn1` gets linted correctly, but `fn2` does not report the unused `y` var.

**Comment by gyandeeps** [comment]
What version of eslint are you using?

**Comment by dominicbarnes** [comment]
```
$ eslint --version
v0.22.1
```

**Comment by nzakas** [comment]
Why is this a bug? `b` is never used in your example?

**Comment by nzakas** [comment]
Oh I see, it's missing `y`. 

Are you using Babel-eslint?

**Comment by dominicbarnes** [comment]
Nah, I'm just using `eslint` core.

**Comment by nzakas** [comment]
Thanks. We should double check that escope is doing the right thing here.

**Comment by yangit** [comment]
````
export default (a, b)=> {
  console.log(a);
};
````

this code still yields no errors.
Version is eslint 1.6.1
Settings as mentioned above

````
{
  "env": {
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "rules": {
    "no-unused-vars": 2
  }
}
````

Expected output is to get a `no-unused-vars` error for `b` variable.

**Comment by ilyavolodin** [comment]
@yangit Could you open a new issues?

**Comment by yangit** [comment]
Done #4047
